libs/community/langchain_community/document_loaders/parsers/grobid.py

```diff
@@ -83,20 +83,20 @@ def process_xml(
                                 "pages": (fpage, lpage),
                             }
                             chunks.append(sentence_dict)
-                    if segment_sentences is not True:
-                        fpage, lpage = (
-                            chunk_bboxes[0][0]["page"],
-                            chunk_bboxes[-1][-1]["page"],
-                        )
-                        paragraph_dict = {
-                            "text": "".join(paragraph_text),
-                            "para": str(i),
-                            "bboxes": chunk_bboxes,
-                            "section_title": sect.text,
-                            "section_number": sect.get("n"),
-                            "pages": (fpage, lpage),
-                        }
-                        chunks.append(paragraph_dict)
+                        if segment_sentences is not True:
```

Comment:
This is what I did when facing this bug. I believe this is enough but I haven't thought too much about it.
I kept the original indentation.
```suggestion
                    if (segment_sentences is not True) and (len(chunk_bboxes) > 0):
```

Reply:
You are right, keeping the original indentation is the right solution. The paragraph_dict needs the full paragraph_text that is build up in the for loop from all individual sentences. Why the chunk_bboxes are empty for some pdfs is another question that I will investigate further given proper internet connection to setup Grobid again.